samples/java-application-with-reflection/pom.xml

```diff
@@ -143,6 +143,7 @@
                             <buildArgs>
                                 <buildArg>--no-fallback</buildArg>
                             </buildArgs>
+                            <agentEnabled>true</agentEnabled>
```

Comment:
I am in favor of naming it `agent` to be consistent with Maven command line (`-Dagent=true`) and Gradle side (`agent = true`).

Reply:
On the Gradle side `agent = true` is only remaining for legacy purposes (backwards compatibility).

The _new_ way to set that is:

```groovy
agent {
    enabled = true
}
```

So I was aligning with that: agent/enabled --> agentEnabled.

Reply:
As an alternative, we could go with an `<agent>` container in the POM, something like this:

```xml
<configuration>
  <agent enabled="true">
    <agentOptions>
      <agentOption><!-- ... --></agentOption
    </agentOptions>
  </agent>
</configuration>
```

Reply:
> ```groovy
> agentOptions {
>     enabled = true
> }
> ```

Though I admit that is confusing.

In the Gradle DSL `agentOptions` is really a container for agent configuration.

Reply:
We could rename to:

```
agentConfiguration {
    enabled = true
    options.add("...")
}
```

if you think that's better. More letters to type, but maybe clearer?

Reply:
If we really want parity with the Gradle DSL, we could even use the "arguments" terminology instead of "options.

Then we might end up with something like this:

```xml
<configuration>
  <agentOptions>
    <enabled>true</enabled>
    <args><!-- "global" agent arguments -->
      <arg><!-- ... --></arg>
    </args>
    <args name="test"><!-- named agent arguments -->
      <arg><!-- ... --></arg>
    </args>
  </agentOptions>
</configuration>
```

Reply:
@melix As just discussed with you, @vjovanov and @sbrannen, let's update the Gradle side with the following idiomatic DSL (your proposal + renaming `agentConfiguration` to `agent`) for clearer semantics to what is defined in https://docs.oracle.com/javase/1.5.0/docs/guide/jvmti/jvmti.html#starting and https://www.graalvm.org/reference-manual/native-image/Agent/ :

```
agent {
    enabled = true
    options.add("...")
}
```

That's a breaking change, but that's IMO reasonable to do it since:
 -  We should not provide suboptimal DSL for backard compatiblity reasons since we are still pre 1.0 (and here we have no choice)
 - This is an advanced feature
 - Migration path is easy and will be documented in released notes

On Maven side, as discussed, let's use:
```
<agent>
    <enabled>true</enabled>
    <options name="main">
        <option>experimental-class-loader-support</option>
    </options>
    <options name="test">
        <option>experimental-class-loader-support</option>
        <option>access-filter-file=${basedir}/src/test/resources/access-filter.json</option>
    </options>
</agent>
```

Reply:
> On Maven side, as discussed, let's use:
> 
> ```xml
> <agent>
>     <enabled>true</enabled>
>     <options name="main">
>         <option>experimental-class-loader-support</option>
>     </options>
>     <options name="test">
>         <option>experimental-class-loader-support</option>
>         <option>access-filter-file=${basedir}/src/test/resources/access-filter.json</option>
>     </options>
> </agent>
> ```

This support has been added in the last two commits.